### PR TITLE
Twice message printing when switching connection

### LINKED ISSUE #772: message is printed twice when switching to the current connection
if moving to the current connection, the message should not appear again: https://github.com/ploomber/jupysql/pull/768#issuecomment-1657862164

### PR DESCRIPTION
**Pull request by anupam-tiwari**
## Issue number

Closes #772 

## Checklist before requesting a review

- [x] Performed a self-review of my code
- [x] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [x] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).

----
:books: Documentation preview :books:: https://jupysql--796.org.readthedocs.build/en/796/

### CODE REVIEW

**Comment by anupam-tiwari** [comment]
@edublancas @neelasha23 does this look good?

File: src/sql/connection/connection.py [hunk 1]
```diff
@@ -205,12 +205,17 @@ def set(
                 existing = rough_dict_get(cls.connections, descriptor)
                 if existing and existing.alias == alias:
                     cls.current = existing
-                # passing an existing descriptor and not alias: use existing connection
                 elif existing and alias is None:
+                    if cls.current != existing:
+                        display.message(
+                            f"Switching to connection {alias or descriptor}"
```
Comment (neelasha23) [inline src/sql/connection/connection.py:211]: This line is repeated 2 times. can be decalred at one place

File: src/tests/test_connection.py [hunk 2]
```diff
@@ -577,7 +588,19 @@ def test_feedback_when_switching_connection_without_alias(ip_empty, tmp_empty, c
     ip_empty.run_cell("%sql duckdb://")
 
     captured = capsys.readouterr()
-    assert "Switching to connection duckdb://" == captured.out.replace("\n", "")
+    assert "Switching to connection duckdb://" == captured.out.splitlines()[-1]
+
+
+def test_feedback_when_switching_connection_with_existing_connection(
+    ip_empty, tmp_empty, capsys
```
Comment (neelasha23) [inline src/tests/test_connection.py:595]: please add a test for the case this message won't be displayed

File: src/sql/connection/connection.py (hunk 1 above)
Comment (anupam-tiwari) [reply src/sql/connection/connection.py:211]: This one is called when we switch from existing connection, the other is  called when a new connection is made

File: src/tests/test_connection.py (hunk 2 above)
Comment (anupam-tiwari) [reply src/tests/test_connection.py:595]: [Added!](https://github.com/ploomber/jupysql/pull/796/files#diff-ecfdf2a1f3adc925d3c32b8aa52540474a10a4160d552f281d7dbd85a0260170R606)

File: src/sql/connection/connection.py (hunk 1 above)
Comment (neelasha23) [reply src/sql/connection/connection.py:211]: I meant the prefix `Switching to connection ` can be declared at one place like 
`SWITCH_PREFIX=Switching to connection` and reused

**Comment by edublancas** [comment]
@anupam-tiwari please fix merge conflicts

**Comment by edublancas** [comment]
@anupam-tiwari is this ready? don't forget to request a review. otherwise, we don't know if the PR is ready for review

File: src/sql/connection/connection.py (hunk 1 above)
Comment (anupam-tiwari) [reply src/sql/connection/connection.py]: [Added!](https://github.com/ploomber/jupysql/pull/796/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R202)

**Comment by anupam-tiwari** [comment]
> @anupam-tiwari is this ready? don't forget to request a review. otherwise, we don't know if the PR is ready for review

@edublancas yes!

**Review by neelasha23 — APPROVED** [review]

**Comment by neelasha23** [comment]
Please resolve merge conflicts @anupam-tiwari

**Review by neelasha23 — CHANGES_REQUESTED** [review]
Please resolve merge conflicts

**Review by edublancas — CHANGES_REQUESTED** [review]
please solve conflicts

also, I noticed that the message appears even when opening a new connection:

```python
In [1]: %load_ext sql

In [2]: %sql duckdb:// --alias a

In [3]: %sql duckdb:// --alias b
Switching to connection b # this should not appear because I'm creating a connection, not switching

In [4]: %sql duckdb:// --alias b

In [5]: %sql duckdb:// --alias a
Switching to connection a

In [6]: %sql duckdb:// --alias a

In [7]:
```

**Comment by anupam-tiwari** [comment]
> please solve conflicts
> 
> also, I noticed that the message appears even when opening a new connection:
> 
> ```python
> In [1]: %load_ext sql
> 
> In [2]: %sql duckdb:// --alias a
> 
> In [3]: %sql duckdb:// --alias b
> Switching to connection b # this should not appear because I'm creating a connection, not switching
> 
> In [4]: %sql duckdb:// --alias b
> 
> In [5]: %sql duckdb:// --alias a
> Switching to connection a
> 
> In [6]: %sql duckdb:// --alias a
> 
> In [7]:
> ```

@edublancas sorry about deleting the previous pr, a little context here: 
Screenshot 2023-08-17 at 12 13 17 PM

since in this case we are creating a brand new connection so I added that case, please let me know if I should modify it!

**Comment by edublancas** [comment]
@anupam-tiwari I don't understand the use case you're talking about in the image and how it is related to my earlier comment, please expand

**Comment by edublancas** [comment]
ok, I think I know what you mean. the image looks like a message from an earlier PR, probably some reviewer's comment (?)

In that case, the message should let the user know that a new connection was created. so that's ok, we can keep the message. but ensure you change it to "created new connection". there's a difference between creating a new connection and switching to an existing one, so ensure that the messages displayed are accurate

**Comment by anupam-tiwari** [comment]
@edublancas since in given image from the closed issue: 
```
In [1]: %load_ext sql

In [2]: %sql duckdb:// 

%%sql duckdb://
CREATE TABLE x1(num INTEGER);

%%sql sqlite://
CREATE TABLE x1(num INTEGER);
Switching to connection sqlite:// # we want this message according to image and caption above it
```
as sqlite:// does now exist previously we are creating a new connection thus I have added the case when a new connection is created the message is displayed [code](https://github.com/ploomber/jupysql/pull/796/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R227-R232), due to this a message is displayed when opening a new connection in your code
```
In [3]: %sql duckdb:// --alias b
Switching to connection b # this should not appear because I'm creating a connection, not switching
```

**Comment by edublancas** [comment]
ok, yeah the message is fine. but let's just ensure the message is accurate. 

if new connection
message: "connecting and switching to {connection}"

if existing connection and just switching
message: "switching to {connection}"

**Comment by anupam-tiwari** [comment]
> ok, yeah the message is fine. but let's just ensure the message is accurate.
> 
> if new connection, then, message: "connecting and switching to {connection}"
> 
> if existing connection and just switching, then, message: "switching to {connection}"

@edublancas [added!](https://github.com/ploomber/jupysql/pull/796/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R242)

File: src/sql/connection/connection.py [hunk 3]
```diff
@@ -208,7 +208,7 @@ def set(
             attributes. As of 0.9.0, only the autocommit option is needed.
         """
         connect_args = connect_args or {}
-        SWITCH_PREFIX = "Switching to connection"
+        SWITCH_PREFIX = "Switching to Connection"
```
Comment (edublancas) [inline src/sql/connection/connection.py:211]: connection shouldn't be capitalized

File: src/tests/test_connection.py [hunk 4]
```diff
@@ -583,7 +583,10 @@ def test_feedback_when_switching_connection_with_descriptors(
     ip_empty.run_cell("%sql sqlite://")
 
     captured = capsys.readouterr()
-    assert "Switching to connection sqlite://" == captured.out.splitlines()[-1]
+    assert (
+        "Connecting and Switching to Connection sqlite://"
```
Comment (edublancas) [inline src/tests/test_connection.py:587]: Switching shouldn't be capitalized

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/tests/test_connection.py (hunk 4 above)
Comment (edublancas) [reply src/tests/test_connection.py:587]: (except when it is the first word in the sentence)

File: src/sql/connection/connection.py (hunk 3 above)
Comment (anupam-tiwari) [reply src/sql/connection/connection.py]: [changed](https://github.com/ploomber/jupysql/pull/796/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R230-R242)

File: src/tests/test_connection.py (hunk 4 above)
Comment (anupam-tiwari) [reply src/tests/test_connection.py]: [changed!](https://github.com/ploomber/jupysql/pull/796/files#diff-62e3d6289e3242b271e6df3a7f201ab003f8ce73a5daf1a824cc2b4d892513d1R230-R242)

**Review by edublancas — APPROVED** [review]

**Pull request merged**